### PR TITLE
Testcases related fixes

### PR DESCRIPTION
**Pull request by rtaneja1**
Removed failing test UpdateUnitFactoryTest. This test depends on the test .nbm files that were removed at the time of donation.
Fixed failing test UpdateProviderFactoryCreateTest.

### CODE REVIEW

File: autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ protected void tearDown () throws  Exception {
     public void testCreate () throws Exception {
         String name = "new-one";
         String displayName = "Newone";
-        URL url = UpdateUnitFactoryTest.class.getResource ("data/catalog.xml");
+        URL url = CatalogCacheTest.class.getResource ("data/catalog.xml");
```
Comment (emilianbold) [inline autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: Why not use `UpdateProviderFactoryCreateTest.class` here? Or even `getClass().getResource()` ?

Comment (rtaneja1) [reply autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: Fixed

File: autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java [hunk 2]
```diff
@@ -50,7 +49,7 @@ protected void tearDown () throws  Exception {
     public void testCreate () throws Exception {
         String name = "new-one";
         String displayName = "Newone";
-        URL url = UpdateUnitFactoryTest.class.getResource ("data/catalog.xml");
+        URL url = getClass().getResource ("/org/netbeans/modules/autoupdate/updateprovider/data/catalog.xml");
```
Comment (emilianbold) [inline autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: Doesn't the relative path data/catalog.xml work anymore?

Comment (rtaneja1) [reply autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: There's no catalog.xml here : https://github.com/apache/incubator-netbeans/tree/master/autoupdate.services/test/unit/src/org/netbeans/api/autoupdate

Comment (rtaneja1) [reply autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: no data/catalog.xml in above location

Comment (rtaneja1) [reply autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: Using the one from https://github.com/apache/incubator-netbeans/tree/master/autoupdate.services/test/unit/src/org/netbeans/modules/autoupdate/updateprovider/data

**Comment by geertjanw** [comment]
What needs to be done to complete this one?

File: autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java (hunk 2 above)
Comment (rtaneja1) [reply autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: Can you please approve this?

Comment (emilianbold) [reply autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: I didn't know you were waiting for me. Sure, it looks good!

Comment (geertjanw) [reply autoupdate.services/test/unit/src/org/netbeans/api/autoupdate/UpdateProviderFactoryCreateTest.java]: OK, will merge this one.

**Comment by geertjanw** [comment]
Merged it.

**Pull request merged**